pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -2408,9 +2409,15 @@ protected void handleAddPartitionToTxn(CommandAddPartitionToTxn command) {
 
         TransactionMetadataStoreService transactionMetadataStoreService =
                 service.pulsar().getTransactionMetadataStoreService();
-        service.pulsar().getTransactionMetadataStoreService().addProducedPartitionToTxn(txnID,
-                command.getPartitionsList())
-                .whenComplete(((v, ex) -> {
+        verifyTxnOwnership(txnID)
```

Comment:
This would require to move the entire logic in the store implementation and pass the current role downstream. I don't think it's a good design because the ServerCnx contains the logic about the current session/role.

getting the TxnMeta is lightweight since it's in memory

Reply:
Is it better if the verification happens after addProducedPartitionToTxn invoke `getTxnMeta `？